File: Triage/bap-0.7-moflow/libtracewrap/libtrace/src/ocaml/utils/copytrace.ml
open Trace_container
open Frame_piqi_ext

let copy_all r w =

  while not r#end_of_trace do
    w#add r#get_frame
  done

let () =
  if Array.length Sys.argv = 3 then
    let r = new reader Sys.argv.(1) in
    let w = new writer ~arch:r#get_arch ~machine:r#get_machine ~frames_per_toc_entry:(r#get_frames_per_toc_entry) Sys.argv.(2) in
    copy_all r w;
    w#finish
  else
    Printf.fprintf stderr "Usage: copytrace <source filename> <destination filename>\n"
;;

File: BAP-0.7-moflow/libtracewrap/libtrace/src/cpp/utils/copytrace.cpp
/** $Id$
 *
 * Copy one trace to another to test API.
 */

#include <fstream>
#include <iostream>
#include "trace.container.hpp"

using namespace SerializedTrace;

void copy_all(TraceContainerReader &r, TraceContainerWriter &w) {
  while (!r.end_of_trace()) {
    w.add(*(r.get_frame()));
  }
}

int main(int argc, char **argv) {
  if (argc != 3) {
    if (argv[0]) {
      std::cout << "Usage: " << argv[0] << " <source filename> <destination filename>" << std::endl;
    }
    exit(1);
  }
  std::string srcfile(argv[1]);
  std::string dstfile(argv[2]);

  TraceContainerReader r(srcfile);
  TraceContainerWriter w(dstfile, r.get_arch(), r.get_machine(), r.get_frames_per_toc_entry(), true);

  copy_all(r, w);
}
